torchgeo/datasets/geo.py

```diff
@@ -569,11 +579,21 @@ def _load_warp_file(self, filepath: str) -> DatasetReader:
         Returns:
             file handle of warped VRT
         """
+        # Todo: need to update meta kwarg `nodata` if it is not set.
+        #  create memory-file?
+        #  https://rasterio.groups.io/g/main/topic/change_the_nodata_value_in_a/28801885?p=
         src = rasterio.open(filepath)
 
         # Only warp if necessary
         if src.crs != self.crs:
-            vrt = WarpedVRT(src, crs=self.crs)
+            valid_nodatavals = [val for val in src.nodatavals if val is not None]
+            if not valid_nodatavals:  # Case for Sentinel2 L1C
+                nodata = 0.0  # Is it safe to assume? For Sentinel2 L1C it is 0.0
+            else:
+                # Get the first valid nodata value.
+                # Usualy the same value for all bands
+                nodata = valid_nodatavals[0]
+            vrt = WarpedVRT(src, nodata=nodata, crs=self.crs)
```

Comment:
Sentinel-2 has as far I can see no value set for nodata. I looked everywhere. Even enabling [alpha-layer](https://gdal.org/drivers/raster/sentinel2.html#open-options) and opening the [MSK_QUALIT-file] in the Sentinel-2 gdal driver, and looking through the (https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-2-msi/level-1c/masks) I found nothing.

Reply:
This change will set the nodata-value. Some datasets have other nodata-values, and we should probably let the user overwrite this, for example in their RasterDataset.

Reply:
Currently, the nodata is only overridden for the warped datasources. The non-warped (already correct CRS) are opened as is, but would also need to have the nodata overridden.